parser/src/string.rs

```diff
@@ -516,7 +522,7 @@ impl<'a> StringParser<'a> {
                     ast::ExprConstant {
                         value: content.into(),
                         kind: None,
-                        range: self.range(),
+                        range: TextRange::new(content_start, self.location),
```

Comment:
Nit: The code so far consistently uses `self.get_pos()` over `location`

```suggestion
                        range: TextRange::new(content_start, self.get_pos()),
```